modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using System;
+using System.Threading.Tasks;
+using Volo.Abp.Application.Dtos;
+using Volo.Abp.Application.Services;
+using Volo.Abp.BlobStoring;
+using Volo.Abp.Content;
+using Volo.Abp.Domain.Repositories;
+using Volo.Abp.GlobalFeatures;
+using Volo.Abp.Users;
+using Volo.CmsKit.Blogs;
+using Volo.CmsKit.GlobalFeatures;
+using Volo.CmsKit.Permissions;
+using Volo.CmsKit.Users;
+
+namespace Volo.CmsKit.Admin.Blogs
+{
+    [RequiresGlobalFeature(typeof(BlogsFeature))]
+    [Authorize(CmsKitAdminPermissions.BlogPosts.Default)]
+    public class BlogPostAdminAppService
+        : CrudAppService<
+            BlogPost,
+            BlogPostDto,
+            Guid,
+            PagedAndSortedResultRequestDto,
+            CreateUpdateBlogPostDto>
+        , IBlogPostAdminAppService
+    {
+        protected readonly IBlogPostManager BlogPostManager;
+        protected readonly IBlogPostRepository BlogPostRepository;
+        protected readonly IBlobContainer<BlogPostCoverImageContainer> BlobContainer;
+        protected readonly ICmsUserLookupService UserLookupService;
+
+        public BlogPostAdminAppService(
+            IRepository<BlogPost, Guid> repository,
+            IBlogPostManager blogPostManager,
+            IBlogPostRepository blogPostRepository,
+            IBlobContainer<BlogPostCoverImageContainer> blobContainer,
+            ICmsUserLookupService userLookupService) : base(repository)
+        {
+            BlogPostManager = blogPostManager;
+            BlogPostRepository = blogPostRepository;
+            BlobContainer = blobContainer;
+            UserLookupService = userLookupService;
+
+            GetListPolicyName = CmsKitAdminPermissions.BlogPosts.Default;
+            GetPolicyName = CmsKitAdminPermissions.BlogPosts.Default;
+            CreatePolicyName = CmsKitAdminPermissions.BlogPosts.Create;
+            UpdatePolicyName = CmsKitAdminPermissions.BlogPosts.Update;
+            DeletePolicyName = CmsKitAdminPermissions.BlogPosts.Delete;
+        }
+
+        public virtual async Task<BlogPostDto> GetByUrlSlugAsync(string urlSlug)
+        {
+            var blogPost = await BlogPostRepository.GetByUrlSlugAsync(urlSlug);
+
+            return MapToGetOutputDto(blogPost);
+        }
+
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Create)]
+        public override async Task<BlogPostDto> CreateAsync(CreateUpdateBlogPostDto input)
+        {
+            _ = await UserLookupService.GetByIdAsync(CurrentUser.GetId());
```

Comment:
Why are we trying to retrieve the user then discard the result?

Reply:
Getting CmsUser over UserLookupService synchronizes CmsUser with Main Application User, and create if not exists.